src/test/java/rx/internal/operators/OperatorGroupByTest.java

```diff
@@ -1804,4 +1809,212 @@ public Integer call(Integer pair) {
         outer.assertValueCount(2);
 
     }
+    
+    @Test
+    public void mapFactoryEvictionWorks() {
+        Func1<Integer, Integer> keySelector = new Func1<Integer, Integer> (){
+            @Override
+            public Integer call(Integer t) {
+                return t /10;
+            }};
+        Func1<Integer, Integer> elementSelector = UtilityFunctions.identity();
+        final List<Integer> evictedKeys = new ArrayList<Integer>();
+        //normally we would use Guava CacheBuilder for instance but to save a test dependency 
```

Comment:
I'll keep the test because i can test expectations of eviction quite precisely but I'll update the comment